### PR TITLE
feat: add react helmet support

### PR DESCRIPTION
**Pull request by reallybeard**
## Description

We are getting more into SEO stuff, and need to render some dynamic titles. So adding react helmet and adding a SEO component to help with that.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #

## Risk

Should cause no functional risk to the user, but possible that a page title is not cleared on page change leaving the wrong title.

### Operations

As you click through sections of the site, the page title should change. This should also update when you change languages.

## Screenshots (if applicable)
Screen_Shot_2022-11-01_at_1 41 40_PM

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -16,17 +16,11 @@
     <link rel="icon" type="image/png" href="%PUBLIC_URL%/favicon.png" integrity="%REACT_APP_SRI_FAVICON_PNG%" crossorigin="anonymous" />
     <link rel="icon" type="image/svg+xml" href="%PUBLIC_URL%/favicon.svg" integrity="%REACT_APP_SRI_FAVICON_SVG%" crossorigin="anonymous" />
     <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no,viewport-fit=cover" />
-    <meta property="og:title" content="ShapeShift DAO">
     <meta property="og:type" content="website" />
-    <meta property="og:description" content="Your Web3 & DeFi Portal">
```
Comment (0xdef1cafe) [inline public/index.html:21]: this will break link previews for discord/twitter etc, as long as you're aware of that

Comment (0xdef1cafe) [reply public/index.html:21]: nvm i see we're doing this below in the seo component

Comment (reallybeard) [reply public/index.html:21]: Yah I guess we can leave this in for when they require SSR

File: public/sitemap.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+https://app.shapeshift.com
+https://app.shapeshift.com/#/connect-wallet
+https://app.shapeshift.com/#/demo/fox
+https://app.shapeshift.com/#/demo/defi
+https://app.shapeshift.com/#/demo/defi/earn
+https://app.shapeshift.com/#/demo/trade
+https://app.shapeshift.com/#/demo/assets
+https://app.shapeshift.com/#/demo/assets/bip122:000000000019d6689c085ae165831e93/slip44:0
+https://app.shapeshift.com/#/demo/assets/eip155:1/slip44:60
```
Comment (0xdef1cafe) [inline public/sitemap.txt:9]: lets chuck in our own token while we're here...

File: src/components/Layout/Seo.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+export const Seo: React.FC<SeoProps> = ({
+  title,
+  description = 'ShapeShift DAO | Your Web3 & DeFi Portal',
+  type = 'website',
+  name = 'ShapeShift',
+}) => {
+  return (
+    <Helmet>
```
Comment (0xdef1cafe) [inline src/components/Layout/Seo.tsx:18]: include `<meta property="og:image" content="https://cloudflare-ipfs.com/ipfs/%REACT_APP_CID_THUMBNAIL_JPG%">` somewhere in here so we don't break link previews

File: src/config.ts [hunk 4]
```diff
@@ -104,6 +104,7 @@ const validators = {
   REACT_APP_KEEPKEY_UPDATER_BASE_URL: url({
     default: 'https://github.com/keepkey/keepkey-updater/releases/download/v2.1.4/',
   }),
+  PUBLIC_URL: url({}),
```
Comment (0xdef1cafe) [inline src/config.ts:107]: we can revert these changes if we're not consuming `getConfig().PUBLIC_URL`

File: src/env/index.ts [hunk 5]
```diff
@@ -3,5 +3,7 @@
 
 // eslint-disable-next-line import/no-default-export
 export default Object.fromEntries(
-  Object.entries(process.env).filter(([k]) => k.startsWith('REACT_APP_')),
+  Object.entries(process.env).filter(
```
Comment (0xdef1cafe) [inline src/env/index.ts:6]: revert

File: src/components/AssetHeader/AssetHeader.tsx [hunk 6]
```diff
@@ -3,8 +3,11 @@ import type { AccountId, AssetId } from '@shapeshiftoss/caip'
 import isEqual from 'lodash/isEqual'
 import { useMemo } from 'react'
 import { AssetIcon } from 'components/AssetIcon'
+import { Seo } from 'components/Layout/Seo'
```
Comment (gomesalexandre) [inline src/components/AssetHeader/AssetHeader.tsx:6]: OCD me would say capitalize this but obviously fine

Comment (reallybeard) [reply src/components/AssetHeader/AssetHeader.tsx:6]: yah it bugs me too, i'll change it real quick.

**Review by gomesalexandre — APPROVED** [review]
Tested locally and does what it says on the box - comments from @0xdef1cafe WRT FOX asset has been tackled and question WRT links preview has been answered
